### PR TITLE
Update issue templates to forms

### PR DESCRIPTION
**Pull request by jennijuju**
Converting lotus issue templates from plain text to forms with validation.

Previews:
- [Bug reports](https://github.com/filecoin-project/lotus/blob/misc/issueform/.github/ISSUE_TEMPLATE/bug-report.yml)
- [Feature request](https://github.com/filecoin-project/lotus/blob/misc/issueform/.github/ISSUE_TEMPLATE/feature_request.yml)
- [Enhancement](https://github.com/filecoin-project/lotus/blob/misc/issueform/.github/ISSUE_TEMPLATE/enhancement.yml)

template links in these three may need updates later.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    description: "Steps to reproduce the behavior"
+    value: |
+      1. Run '...'
+      2. Do '...'
+      3. See error '...'
+      ...
+    render: bash
+  validations:
+    required: true
+- type: textarea
```
Comment (Stebalien) [inline .github/ISSUE_TEMPLATE/bug-report.yml:104]: Given the "describe the bug" section, this seems redundant.

File: .github/ISSUE_TEMPLATE/bug-report.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    - lotus client
+    - lotus market
+    - lotus JSON-RPC API
+    - Other
+- type: dropdown
+  id: area
+  validations: 
+    required: true
+  attributes:
+    label: Area
```
Comment (Stebalien) [inline .github/ISSUE_TEMPLATE/bug-report.yml:43]: This seems to overlap with the component section a bit. Personally, I'd just combine them.

File: .github/ISSUE_TEMPLATE/enhancement.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    options:
+    - syncing
+    - sealing
+    - mining (WinningPoSt)
+    - proving (WindowPoSt)
+    - deal making
+    - api call
+    - message management (mpool)
+    - Other
+- type: textarea
```
Comment (Stebalien) [inline .github/ISSUE_TEMPLATE/enhancement.yml:48]: I'd opt for a single free-form section, honestly. Requiring structure can help in some cases, but I usually find it just leads to redundant or pointless text.

Comment (jennijuju) [reply .github/ISSUE_TEMPLATE/enhancement.yml:48]: updated!

File: .github/ISSUE_TEMPLATE/bug-report.yml (hunk 2 above)
Comment (jennijuju) [reply .github/ISSUE_TEMPLATE/bug-report.yml]: combined

**Pull request merged**